uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/apis/work-pattern-api-test.js

```diff
@@ -0,0 +1,46 @@
+define([
+  'mocks/data/work-pattern-data',
+  'leave-absences/shared/apis/work-pattern-api',
+], function (mockData) {
+  'use strict';
+
+  describe('WorkPatternAPI', function () {
+    var WorkPatternAPI, $httpBackend;
+
+    beforeEach(module('leave-absences.apis'));
+
+    beforeEach(inject(function (_WorkPatternAPI_, _$httpBackend_) {
+      WorkPatternAPI = _WorkPatternAPI_;
+      $httpBackend = _$httpBackend_;
+
+      //Intercept backend calls for WorkPatternAPI.getCalendar
+      $httpBackend.whenGET(/action\=getcalendar&entity\=WorkPattern/)
+        .respond(mockData.calenderData());
+    }));
+
+    describe('calenderData()', function () {
+      var workPatternPromise;
+
+      beforeEach(function () {
+        spyOn(WorkPatternAPI, 'sendGET').and.callThrough();
+        workPatternPromise = WorkPatternAPI.getCalendar();
```

Comment:
This is not representative of how you use the method. You would never call `getCalendar()` without the contact id and period id (the api endpoint would throw an error anyway, those parameters are mandatory!), so why are you doing it in the test?